src/derived/slot.rs

```diff
@@ -429,55 +429,29 @@ where
             self, revision, revision_now,
         );
 
-        // Acquire read lock to start. In some of the arms below, we
-        // drop this explicitly.
-        let state = self.state.read();
+        // Check the current state and potentially block if there's a computation in progress.
+        let state = match self.probe(db, self.state.read(), runtime, revision_now) {
```

Comment:
So the problem here is that we are going to clone the value in this case, which isn't necessary

Reply:
Indeed! Thanks for catching this.